flake.nix

```diff
@@ -47,8 +74,59 @@
+                  schema-profunctor = hself.callPackage "${inputs.wire-server}/libs/schema-profunctor" {inherit gitignoreSource;};
+                  sodium-crypto-sign = hself.callPackage "${inputs.wire-server}/libs/sodium-crypto-sign" {inherit gitignoreSource;};
+                  types-common = hself.callPackage "${inputs.wire-server}/libs/types-common" {inherit gitignoreSource;};
+                  wai-utilities = hself.callPackage "${inputs.wire-server}/libs/wai-utilities" {inherit gitignoreSource;};
+                  wire-api =
+                    addBuildDepends [config.packages.mls-test-cli]
+                    (hself.callPackage "${inputs.wire-server}/libs/wire-api" {inherit gitignoreSource;});
+                  wire-message-proto-lens =
+                    addBuildTools [pkgs.protobuf]
+                    (hself.callPackage "${inputs.wire-server}/libs/wire-message-proto-lens" {inherit gitignoreSource;});
+                  zauth = hself.callPackage "${inputs.wire-server}/libs/zauth" {inherit gitignoreSource;};
+
+                  # libraries maintained in external repositories
+                  amqp = dontCheck (hself.callCabal2nix "amqp" "${inputs.amqp}" {});
+                  cql-io = (hself.callCabal2nix "cql-io" "${inputs.cql-io}" {}).overrideAttrs {doCheck = false;};
+                  ghc-source-gen = hself.callCabal2nix "ghc-source-gen" "${inputs.ghc-source-gen}" {};
+                  hsaml2 = (hself.callCabal2nix "hsaml2" "${inputs.hsaml2}" {}).overrideAttrs {doCheck = false;};
+                  http-client = hself.callCabal2nix "http-client" "${inputs.http-client}/http-client" {};
+                  http-client-openssl = hself.callCabal2nix "http-client" "${inputs.http-client}/http-client-openssl" {};
+                  http-client-tls = hself.callCabal2nix "http-client" "${inputs.http-client}/http-client-tls" {};
+                  http-conduit = hself.callCabal2nix "http-client" "${inputs.http-client}/http-conduit" {};
+                  saml2-web-sso = dontCheck (hself.callCabal2nix "saml2-web-sso" "${inputs.saml2-web-sso}" {});
+                  tinylog = hself.callCabal2nix "tinylog" "${inputs.tinylog}" {};
+                  transitive-anns = (hself.callCabal2nix "transitive-anns" "${inputs.transitive-anns}" {}).overrideAttrs {doCheck = false;};
+                  wai-predicates = hself.callCabal2nix "wai-predicates" "${inputs.wai-predicates}" {};
+                  wai-routing = doJailbreak (hself.callCabal2nix "wait-routing" "${inputs.wai-routing}" {});
+
+                  # libraries from hackage
+                  HsOpenSSL = hself.callHackage "HsOpenSSL" "0.11.7.6" {};
+                  wai-route = hself.callHackage "wai-route" "0.4.0" {};
+                  warp = dontCheck (hself.callHackage "warp" "3.3.29" {});
+                  warp-tls = hself.callHackage "warp-tls" "3.4.3" {};
+                  tls = hself.callHackage "tls" "1.9.0" {};
```

Comment:
these are pretty much the ones we have to "maintain", the rest of it will be pretty much automatic. Do you think that's that problematic, @smatting?